root/etc/cont-init.d/94-ros2-src

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/with-contenv bash
+
+echo "**** installing ros2 dev environment ****"
+if ! dpkg -l | grep gnupg > /dev/null; then
+    apt-get update && apt-get install -y gnupg
+fi
+
+source /etc/lsb-release
+[[ ! -f "/etc/apt/sources.list.d/ros2.list" ]] && \
```

Comment:
Please change this to an if block. Otherwise the container will fail on a subsequent start because this part will exit 1 and since it's the last line in this init script, the init script will exit 1, failing the whole init

Reply:
I changed this to an if block and moved the ros2 sourcing into the `.bashrc` inside the home folder.